common/schema-builder/src/main/java/io/smallrye/graphql/schema/helper/BeanValidationDirectivesHelper.java

```diff
@@ -91,6 +101,32 @@ public List<DirectiveInstance> transformBeanValidationConstraintsToDirectives(An
                 directive.setValue("min", getLongValue(annotations, annotationName, "value"));
                 result.add(directive);
             }
+            if (annotationName.equals(VALIDATION_ANNOTATION_DECIMAL_MAX)) {
+                DirectiveInstance directive = new DirectiveInstance();
+                directive.setType(CONSTRAINT_DIRECTIVE_TYPE);
+                String value = getStringValue(annotations, annotationName, "value");
+                try {
+                    directive.setValue("maxFloat", new BigDecimal(value).doubleValue());
+                    result.add(directive);
+                } catch (NumberFormatException nfe) {
+                    LOGGER.warn(
+                            "Not generating a bean validation directive for " + annotations.getAnnotationValue(annotationName) +
+                                    " because the value can't be parsed as a BigDecimal");
+                }
```

Comment:
Maybe we could extract a `getDoubleValue` and add a constructor to `DirectiveInstance`, which would help the other directives, too?

```suggestion
                double value = getDoubleValue(annotations, annotationName, "value");
                result.add(new DirectiveInstance(CONSTRAINT_DIRECTIVE_TYPE, "maxFloat", value));
```

Reply:
For `DirectiveInstance` I'd prefer sticking to using the default constructor and setters, because Quarkus bytecode recorder uses just that, so the code is closer to the actual bytecode that will be generated by Quarkus. Also the constructor would have to accept a Map of values for cases where we have multiple values, so it wouldn't be all that nice anyway.